File: src/TIL_0721/Solution_d3_7510_연속합.py
T = int(input())

for tc in range(1, T+1):
    N = int(input())
    cnt = 1
    for i in range(1, N//2+1):
        next = i
        n = N-i
        while n>0:
            next += 1
            n -= next
        if n==0:
            cnt += 1        
    print(f'#{tc} {cnt}') 

File: src/TIL_0721/Solution_d3_7510_연속합.java
package TIL_0721;
import java.util.*;

public class Solution_d3_7510_연속합 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
		for(int t=1; t<=T; t++) {
			int N = sc.nextInt();
			int cnt = 1;
			for(int i=1; i<N/2+1; i++) {  
				int next = i;
				int n = N - i;
				while(n > 0) {
					next++;
					n -= next;
				}
				if(n == 0) {
					cnt++;
				}
			}
			System.out.println("#" + t + " " + cnt);
				
		}
	}//end of main
}//end of class
